### PR TITLE
Added gluster cluster status calculation logic

### PR DESCRIPTION
**Pull request by shtripat**
Signed-off-by: Shubhendu <shtripat@redhat.com>

### CODE REVIEW

**Comment by shtripat** [comment]
This creates the status details in central store as below 

```
{

    "action": "get",
    "node": {
        "key": "/clusters/f4b4ea0d-25a7-459c-a19c-14460041d23f/GlobalDetails/status",
        "value": "healthy",
        "modifiedIndex": 4165,
        "createdIndex": 4165
    }
}
```

**Comment by shtripat** [comment]
@Tendrl/tendrl-core plz review

File: etc/tendrl/gluster-integration/gluster-integration.conf.yaml.sample [hunk 1]
```diff
@@ -4,3 +4,4 @@ tendrl_exe_file_prefix: $HOME/.tendrl/util_runner_
 log_cfg_path: /etc/tendrl/gluster-integration/gluster-integration_logging.yaml
 log_level: DEBUG
 logging_socket_path: /var/run/tendrl/.message.sock
+snapshot_support: 3.6
```
Comment (r0h4n) [inline etc/tendrl/gluster-integration/gluster-integration.conf.yaml.sample]: what is this flag? Add a comment here

Comment (shtripat) [inline etc/tendrl/gluster-integration/gluster-integration.conf.yaml.sample]: Removed now as we would be using gstatus as is.

File: requirements.txt [hunk 2]
```diff
@@ -7,4 +7,5 @@ python-etcd
 python-dateutil
 gevent>=1.0
 greenlet>=0.3.2
+gstatus
```
Comment (shtripat) [inline requirements.txt]: We need to build gstatus on our own and I have added dependency already here.
@TimothyAsirJeyasing plz do the needful.

File: tendrl/gluster_integration/sds_sync/__init__.py [hunk 3]
```diff
@@ -24,6 +27,22 @@ def _run(self):
...
+                args = ["gstatus", "-o", "json"]
+                p = subprocess.Popen(
+                    args,
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.PIPE,
+                    stdin=open(os.devnull, "r")
+                )
+                stdout, stderr = p.communicate()
+                if stderr == "":
+                    out_dict = json.loads(stdout[stdout.index('{'): -1])
```
Comment (r0h4n) [inline tendrl/gluster_integration/sds_sync/__init__.py]: Please provide output of "gstatus -o json"

**Pull request merged**